mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -33,85 +35,104 @@ class RenderAssetGridElement {
   });
 }
 
-List<RenderAssetGridElement> assetsToRenderList(
-  List<Asset> assets,
-  int assetsPerRow,
-) {
-  List<RenderAssetGridElement> elements = [];
-
-  int cursor = 0;
-  while (cursor < assets.length) {
-    int rowElements = min(assets.length - cursor, assetsPerRow);
-    final date = assets[cursor].createdAt;
-
-    final rowElement = RenderAssetGridElement(
-      RenderAssetGridElementType.assetRow,
-      date: date,
-      assetRow: RenderAssetGridRow(
-        assets.sublist(cursor, cursor + rowElements),
-      ),
-    );
-
-    elements.add(rowElement);
-    cursor += rowElements;
-  }
-
-  return elements;
-}
-
-List<RenderAssetGridElement> assetGroupsToRenderList(
+/// Converts assets grouped by date to the data model required by Immich's custom asset grid implementation.
+Future<List<RenderAssetGridElement>> assetGroupsToRenderList(
   Map<String, List<Asset>> assetGroups,
   int assetsPerRow,
-) {
-  List<RenderAssetGridElement> elements = [];
-  DateTime? lastDate;
-
-  assetGroups.forEach((groupName, assets) {
-    try {
-      final date = DateTime.parse(groupName);
+) async {
+
+  processData(Map<String, Object> data) {
+    final monthFormat = DateFormat(data["month_format"]! as String);
```

Comment:
just thinking outloud here, is there a way to make this into a class/data structure so we can gain the benefit of strong type inference?

Reply:
Oh yeah, sure. I always wanted to do that but kind of forgot about it :D